libr/anal/reflines.c

```diff
@@ -122,12 +122,32 @@ R_API RList *r_anal_reflines_get(RAnal *anal, ut64 addr, const ut8 *buf, ut64 le
 		}
 		addr += sz;
 		{
-			RAnalMetaItem *mi = r_meta_find_any_except (anal, addr, R_META_TYPE_COMMENT, 0);
-			if (mi) {
-				ptr += mi->size;
-				addr += mi->size;
-				r_meta_item_free (mi);
-				goto __next;
+			RPVector *metas = r_meta_get_all_at (anal, addr);
+			if (metas) {
+				void **it;
+				ut64 skip = 0;
+				r_pvector_foreach (metas, it) {
+					RIntervalNode *node = *it;
+					RAnalMetaItem *meta = node->data;
+					switch (meta->type) {
```

Comment:
isn't it better to do
```C
switch (meta->type) {
case R_META_COMMENT:
   break;
default:
  skip = ..
  goto do_skip;
}
```
It seems to me it would be more in line with the old semantic.

Reply:
In theory yes, but after the refactoring all metas have a size of at least 1 (i.e. they cover at least one byte. Before they could have 0). This can lead to some brokenness here with the types not listed in the switch.